### PR TITLE
TT-69 Isomorphic Integration Test Deployments

### PR DESCRIPTION
**Pull request by tateexon**
Updates smoke and soak tests to use the isomorphic deployment strategy for CI.
Reuses nix for the test image build.
In CI build the contracts in parallel to the chainlink image to speed up the pipeline.
Use larger runners to reduce build time.
Update nix to add missing pip and helm chart dependencies.
Add make calls to run nix in a container if it is not behaving on mac.

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -35,7 +36,9 @@
   venvDir = "./.venv";
 
   postShellHook = ''
-    pip install --upgrade pip
     pip install -r ${./contracts/requirements.txt} -c ${./contracts/constraints.txt}
+    helm repo add chainlink-qa https://raw.githubusercontent.com/smartcontractkit/qa-charts/gh-pages/
+    helm repo add bitnami https://charts.bitnami.com/bitnami
+    helm repo update
```
Comment (archseer) [inline shell.nix]: An idea: add the `~/.config/helm/repositories.yaml` file to the project, then set `HELM_REPOSITORY_CONFIG` in this `shell.nix` to point to the file. That way it's part of the project and doesn't require running shell hooks

File: integration-tests/test.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM nixos/nix:latest
```
Comment (archseer) [inline integration-tests/test.Dockerfile:1]: ❤️

**Comment by tateexon** [comment]
Well got the e2e tests to pass but somehow broke the contracts tests that I didn't think I touched in any way? Gonna have to figure that out tomorrow.

File: shell.nix (hunk 1 above)
Comment (tateexon) [reply shell.nix]: Still had to leave in helm repo update or we wouldn't pull in the cache.

**Comment by archseer** [comment]
Contract tests seem to have broken on their own with a devnet update I think, it's now broken on other branches/master

**Review by kalverra — APPROVED** [review]

**Pull request merged**